src/org/exist/storage/BrokerPools.java

```diff
@@ -0,0 +1,319 @@
+     *
+     * @throws EXistException If the initialization fails.
+     *
+     * @Deprecated Use {@link #configure(String, int, int, Configuration, Optional)}
+     */
+    @Deprecated
+    public static void configure(final String instanceName, final int minBrokers, final int maxBrokers,
+            final Configuration config) throws EXistException {
+        configure(instanceName, minBrokers, maxBrokers, config, Optional.empty());
+    }
+
+    /**
+     * Creates and configures a database instance and adds it to the pool.
+     * Call this before calling {link #getInstance()}.
+     * If a database instance with the same name already exists, the new configuration is ignored.
+     *
+     * @param instanceName A <strong>unique</strong> name for the database instance.
+     *                     It is possible to have more than one database instance (with different configurations
+     *                     for example).
+     * @param minBrokers        The minimum number of concurrent brokers for handling requests on the database instance.
+     * @param maxBrokers        The maximum number of concurrent brokers for handling requests on the database instance.
+     * @param config            The configuration object for the database instance
+     * @param statusObserver    Observes the status of this database instance
+     *
+     * @throws EXistException If the initialization fails.
+     */
+    public static void configure(final String instanceName, final int minBrokers, final int maxBrokers,
+            final Configuration config, final Optional<Observer> statusObserver) throws EXistException {
+
+        // optimize for read-concurrency as instances are configured (created) once and used many times
+        final Lock readLock = instancesLock.readLock();
+        readLock.lock();
+        try {
+            if (instances.containsKey(instanceName)) {
+                LOG.warn("Database instance '" + instanceName + "' is already configured");
+                return;
+            }
+        } finally {
+            readLock.unlock();
+        }
+
+        // fallback to probably having to create a new BrokerPool instance
+        final Lock writeLock = instancesLock.writeLock();
+        writeLock.lock();
+        try {
+            // check again, as another thread may have pre-empted us since we released the read-lock
+            if (instances.containsKey(instanceName)) {
+                LOG.warn("Database instance '" + instanceName + "' is already configured");
+                return;
+            }
+
+            LOG.debug("Configuring database instance '" + instanceName + "'...");
+            try {
+                //Create the instance
+                final BrokerPool instance = new BrokerPool(instanceName, minBrokers, maxBrokers, config, statusObserver);
+                //Add it to the list
+                instances.put(instanceName, instance);
```

Comment:
@shabanovd Okay I see the issue for `Realm`, although it doesn't stop the Realm from starting (as it will subsequently be started by the SecurityMangerImpl), so in reality all works well.

I would ask, in general, should the `Configurator` really be starting sub-modules???

Also it seems this is a mess as a `Realm` would before this PR have been started twice (unnecessarily):

1. Line 688 of `Configurator`:
    ```java
    if (db != null) {
        try(final DBBroker broker = db.getBroker()) {
            ((LifeCycle) obj).start(broker);
        }
    }
```

2. Line 200 of `SecurityManagerImpl`:
    ```java
        for (final Realm realm : realms) {
            realm.start(broker);
        }
```